platform/linux-generic/odp_packet.c

```diff
@@ -1739,9 +1739,68 @@ void odp_packet_print(odp_packet_t pkt)
 		seg = odp_packet_next_seg(pkt, seg);
 	}
 
-	str[len] = '\0';
+	ODP_PRINT("%s\n", str);
+}
+
+void odp_packet_print_data(odp_packet_t pkt, uint32_t offset,
+			   uint32_t byte_len)
+{
+	odp_packet_hdr_t *hdr = packet_hdr(pkt);
+	uint32_t bytes_per_row = 16;
+	int num_rows = (byte_len + bytes_per_row - 1) / bytes_per_row;
+	int max_len = 256 + (3 * byte_len) + (3 * num_rows);
+	char str[max_len];
+	int len = 0;
+	int n = max_len - 1;
+	uint32_t data_len = odp_packet_len(pkt);
+	pool_t *pool = hdr->buf_hdr.pool_ptr;
+
+	len += snprintf(&str[len], n - len, "Packet\n------\n");
+	len += snprintf(&str[len], n - len,
+			"  pool index    %" PRIu32 "\n", pool->pool_idx);
```

Comment:
My understanding is that `PRIu32/PRIu16` are not required, you can always use `%u/%hu` instead. Only `PRIu64/PRIx64` are required, because exact `uint64_t` definition depends on platform ABI.

Reply:
we had some problems with that prints in the past. Arm build or -m32 do not remember exactly what it was.

Reply:
@muvarov could you please be more specific? What kind of problems?

Reply:
@lumag  compilation errors. So we have comits:
 27a05a1d61f3b7e4d5

42d0b99c453f269f

d25d276c649f57021f

Reply:
It's never wrong to use the PRIxxx macros so that's the safest and most portable way to go. So I'm happy to see them used here and we should try to use them in new development that calls for these sort of conversions.

Reply:
@muvarov they do not mention any compiler errors, so they are useless in trying to understand, which errors were there.

Reply:
@lumag  like here: https://answers.launchpad.net/gcc-arm-embedded/+question/293823 old gcc  looks like failed. But int usually  = uint32_t, so it can work on both cases. I prefer see exact type.